app/javascript/member-facing/double_opt_in.js

```diff
@@ -21,31 +21,21 @@ const showNotice = () => {
 };
 
 const DoubleOptIn = {
-  version(versionNumber?: number) {
-    if (versionNumber) {
-      version = versionNumber;
-      $('.action-form').append(
-        `<input type='hidden' name='test_version' value='${version}' />`
-      );
-    }
-    return version;
-  },
-
   handleActionSuccess(resp: Response) {
     if (!resp || !resp.double_opt_in) return;
 
-    if (DoubleOptIn.version() === 2) {
-      showNotice();
-    } else {
-      window.location.href = resp.follow_up;
-    }
+    showNotice();
   },
 };
 
-$(() =>
-  $('.action-form').on('ajax:success', (e, data) =>
-    DoubleOptIn.handleActionSuccess(data)
-  )
-);
+$(() => {
+  if (window.location.search.match(/double_opt_in=true/)) {
+    showNotice();
+  }
+
+  $('.action-form').on('ajax:success', (e, data) => {
+    DoubleOptIn.handleActionSuccess(data);
```

Comment:
I still think that the event listener is too broad. Why isn't it in the petition backbone component?

Reply:
Cheers, @eyko. Now fixed.